### ISSUE TITLE
Don't write logs to the dir at cwd by default

### ISSUE THREAD

**Issue by matklad** [rust-analyzer/rust-analyzer#282]
[This](https://github.com/rust-analyzer/rust-analyzer/blob/1875aa024f885f1d6119bff8aa828634b4941248/crates/ra_lsp_server/src/main.rs#L9-L13) writes the logs to the dir at cwd unconditionally. This is useful for development, but can trip new users if the cwd is not writable for some reasons.

Let's add an `RA_INTERNAL_MODE` env var to control logging behavior. By default we log only to stderr, but with this var set we revert to the current behavior.

**Comment by eminence** [comment]
Maybe if the envvar is set, it could name a directory into which logs are written?

**Comment by matklad** [comment]
That would be a possible extension, yeah, but I'd just start with a simple "is var equal to `1`" check: I wouldn't use this feature personally, for me the cwd is most convenient.

**Comment by eminence** [comment]
On my Windows machine, the cwd is `C:\Program Files\Microsoft VS Code`

**Comment by CAD97** [comment]
I suppose cwd also depends on whether you launch code from the command line (with `code .` or `code path/to/project`) versus using a shortcut (my start menu shortcut is "Start in: `"S:\Program Files\Microsoft VS Code"`).

TL;DR cwd of a GUI program is messy, and it seems the cwd of the language server inherits that.
